### DISCUSSION TITLE
[Feature] Separate photo and video upload

### DISCUSSION THREAD

**Discussion by Pro100x3mal** [immich-app/immich#3341] [Feature Request]
### The feature

Please add an option in the mobile application that will allow you to upload photos and videos separately to the server. So that you can start uploading all photos first and then, for example, at night, heavy video files

### Platform

- [X] Server
- [ ] Web
- [X] Mobile

**Comment by klejejs** [comment]
Yes please, this is huge.
Without it, background backup for me on mobile (iOS) is useless as it always tries to upload large videos, doesn't finish and gets stuck in the loop.

**Reply by klejejs** [reply]
What Google Photos does in this regard is sort photos first and videos after on mobile. Maybe this could be much easier to implement.

**Reply by Phyks** [reply]
Although prioritizing photos this way would at least ensure that photos upload is not blocked by pending videos, it would not fully address the issue. Videos would still be in the queue and be retried over and over, which under roaming (relying on mobile data and moving around, not speaking here about limited data) circumstances would still be an issue. The network connectivity would likely be lost in the upload of the large files, making the upload fail and be retried, so likely still having a huge impact on battery.

**Comment by varchasgopalaswamy** [comment]
I don't know enough dart to test or implement this, but it seems like the function `buildUploadCandidates` in `backup.service.dart` would need to be modified to achieve this. `buildUploadCandidates` returns a set of `BackupCandidate`,  and `BackupCandidate.asset.durationInSeconds` is, if I understand right, the length of the video (0 if an image). I guess converting the set to a list, sorting by `BackupCandidate.asset.durationInSeconds`, then converting back to a set should achieve this?

**Comment by bo0tzz** [comment]
#3120
